### DISCUSSION TITLE
How to simulate a fake PR?

### DISCUSSION THREAD

**Discussion by jrson83** [cardinalby/github-action-ts-run-api#3] [Q&A] [answered]
My action queries the last PR with `pulls.listFiles` and then assigns a label to the issue with `issues.addLabels`.

I have read the documentation several times and looked at many examples which I found through the github search.  I can't figure out, how can I create a fake PR to use it with the tests. Is it possible to simulate or create a fake pull request before the action test runs?

I [created a gist](https://gist.github.com/jrson83/db3e30952c086ed3ce54ff661aa755f8) with the code I have tried. If you have any suggestions I would be pleased.

**Comment by cardinalby** [comment]
Hi! Looked quickly at your gist, I believe you chose the right approach with mocking API. Creating and then deleting a temporary PR using API can be also an option. Will look at it during the week. Can you compose your gist to a repo so that I can test it locally?

**Reply by jrson83** [reply]
Thanks! Yeah sure, I will do and update the link here.

**Comment by cardinalby** [comment, marked as answer]
Hi! Is it still relevant?

I came with 2 suggestions regarding your need.

## First
You can manually create a test PR, look at its ID and put this ID into options githubContext:
```js
options.setGithubContext({ payload: { pull_request: { number: YOUR_HARDCODED_ID } } })
```
This way you don't need to mock API. 

You can even use a separate repo if you don't like bloating your main repo. In this case use 
```js
options.setGithubContext({ 
    payload: { pull_request: { number: YOUR_HARDCODED_ID } },
    repository: 'owner/repo'
})
```

## Second
You can extract the code of your action into a function and call it passing API client:

```js
export async function runAction(apiClient) { 
   ... 
}

runAction(new Octokit({ auth: token })
```

And use [Single Function Target](https://github.com/cardinalby/github-action-ts-run-api/blob/master/docs/run-targets/function.md) to run the function in an emulated Action environment. This way you can mock _ApiClient_ by normal jest mocks:

```js
import {runAction} from 'main.js'

const apiMock = createJestMock()

const target = RunTarget.asyncFn(
    async () => runAction(apiMock), 
    'path/to/action.yml'
);
const result = target.run(RunOptions.create( ... ))
```

**Reply by jrson83** [reply]
Sorry I didn't reply. Thank you very much I will take a look at it again.
